arches/app/views/api/controlled_lists.py

```diff
@@ -0,0 +1,424 @@
+import logging
+from http import HTTPStatus
+from uuid import UUID
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.db.models import Max
+from django.views.generic import View
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+
+from arches.app.models.models import (
+    ControlledList,
+    ControlledListItem,
+    ControlledListItemImage,
+    ControlledListItemImageMetadata,
+    ControlledListItemValue,
+    Language,
+    Node,
+)
+from arches.app.models.utils import field_names
+from arches.app.utils.betterJSONSerializer import JSONDeserializer
+from arches.app.utils.decorators import group_required
+from arches.app.utils.permission_backend import get_nodegroups_by_perm
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.utils.string_utils import str_to_bool
+
+logger = logging.getLogger(__name__)
+
+
+def _prefetch_terms(request):
```

Comment:
I get leaving this here an I'm ok with it. Alternatively it could be moved to a util to keep the API simple, but I do not feel very strongly about this 👍